cpp/core/config/GlutenConfig.cc

```diff
@@ -18,40 +18,21 @@
 #include <jni.h>
 
 #include "compute/ProtobufUtils.h"
+#include "config.pb.h"
 #include "jni/JniError.h"
-#include "substrait/plan.pb.h"
 
 namespace gluten {
-std::unordered_map<std::string, std::string> getConfMap(JNIEnv* env, jbyteArray planArray) {
+std::unordered_map<std::string, std::string> parseConfMap(JNIEnv* env, jbyteArray configArray) {
   std::unordered_map<std::string, std::string> sparkConfs;
-  auto planData = reinterpret_cast<const uint8_t*>(env->GetByteArrayElements(planArray, 0));
-  auto planSize = env->GetArrayLength(planArray);
-  ::substrait::Plan subPlan;
-  gluten::parseProtobuf(planData, planSize, &subPlan);
+  auto planData = reinterpret_cast<const uint8_t*>(env->GetByteArrayElements(configArray, 0));
+  auto planSize = env->GetArrayLength(configArray);
+  ConfigMap pConfigMap;
+  gluten::parseProtobuf(planData, planSize, &pConfigMap);
 
-  if (subPlan.has_advanced_extensions()) {
-    auto extension = subPlan.advanced_extensions();
-    if (extension.has_enhancement()) {
-      const auto& enhancement = extension.enhancement();
-      ::substrait::Expression expression;
-      if (!enhancement.UnpackTo(&expression)) {
-        std::string errorMessage =
-            "Can't Unapck the Any object to Expression Literal when passing the spark conf to velox";
-        throw gluten::GlutenException(errorMessage);
-      }
-      if (expression.has_literal()) {
-        auto literal = expression.literal();
-        if (literal.has_map()) {
-          auto literalMap = literal.map();
-          auto size = literalMap.key_values_size();
-          for (auto i = 0; i < size; i++) {
-            ::substrait::Expression_Literal_Map_KeyValue keyValue = literalMap.key_values(i);
-            sparkConfs.emplace(keyValue.key().string(), keyValue.value().string());
-          }
-        }
-      }
-    }
+  for (const auto& pair : pConfigMap.configs()) {
+    sparkConfs.emplace(pair.key(), pair.value());
```

Comment:
Should we add a check for duplicated keys?

Reply:
Generated map from java side have guarantee this.